### PR TITLE
Remove unnecessary options structs

### PR DESCRIPTION
**Pull request by chlowell**
Part of #380. This removes the options bags for methods and constructors from the public API. These types are useless in application code because they aren't input to any function (applications set options by calling functions like `WithAuthority()` instead). They could conceivably be used in unusual test scenarios as part of a fake MSAL however, because these types distract from the real options API, I think we shouldn't export them before seeing a clear need for them; doing so would be non-breaking.

### CODE REVIEW

File: apps/confidential/confidential.go [hunk 1]
```diff
@@ -243,97 +243,74 @@ func AutoDetectRegion() string {
 // package doc. A new Client should be created PER SERVICE USER.
 // For more information, visit https://docs.microsoft.com/azure/active-directory/develop/msal-client-applications
 type Client struct {
-	base base.Client
-
-	cred *accesstokens.Credential
-
+	accessor                          cache.ExportReplace
```
Comment (bgavrilMS) [inline apps/confidential/confidential.go:246]: should we call it more in the lines of `cache serialization` ? Maybe `cacheSerializer` ?

Comment (bgavrilMS) [reply apps/confidential/confidential.go:246]: (not sure if this is part of public API. If not, ignore comment)

File: apps/public/public.go [hunk 2]
```diff
@@ -47,101 +48,87 @@ type AuthResult = base.AuthResult
...
+	return func(c *Client) {
+		c.disableInstanceDiscovery = !enabled
 	}
 }
 
 // Client is a representation of authentication client for public applications as defined in the
 // package doc. For more information, visit https://docs.microsoft.com/azure/active-directory/develop/msal-client-applications.
 type Client struct {
-	base base.Client
+	accessor                 cache.ExportReplace
```
Comment (bgavrilMS) [inline apps/public/public.go:105]: most of these are in common with confidential client.

**Review by bgavrilMS — APPROVED** [review]

File: apps/confidential/confidential.go (hunk 1 above)
Comment (chlowell) [reply apps/confidential/confidential.go:246]: Just FYI, names beginning with a lowercase letter are private

**Review by jhendrixMSFT — APPROVED** [review]

File: apps/public/public.go (hunk 2 above)
Comment (chlowell) [reply apps/public/public.go]: There are ways we could share these fields but I haven't thought of one that's really better than duplicating them. We could for example move the common fields onto a new type and have constructor options set them there. However, accommodating options not shared between client types would then be difficult and more complex than duplicating the common fields.

**Pull request merged**